pkg/config/validation.go

```diff
@@ -0,0 +1,347 @@
+package config
+
+import (
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/ghodss/yaml"
+	"github.com/loft-sh/vcluster/config"
+	"github.com/loft-sh/vcluster/pkg/util/toleration"
+	"github.com/samber/lo"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+)
+
+var allowedPodSecurityStandards = map[string]bool{
+	"privileged": true,
+	"baseline":   true,
+	"restricted": true,
+}
+
+var (
+	verbs = []string{"get", "list", "create", "update", "patch", "watch", "delete", "deletecollection"}
+)
```

Comment:
you can add enums with jsonschema using tags so that will even work within the helm chart

Reply:
We need this within golang code as well, e.g. if people don't use our helm chart

Reply:
but now if they do only use our helm charts they don't have the validation
We can run jsonschema validator within go and get the errors, that way validation would be consistent across the board

Reply:
https://github.com/santhosh-tekuri/jsonschema/tree/v5.3.1 for example would allow us to validate the schema for everything that we can, and then rely on go code to only validate things that aren't possible with the schema
That also would allow us to be able to validate for more than 1 version when we introduce breaking changes